### PR TITLE
Now supports extracting all objects and optionally retain coordinates

### PR DESCRIPTION
**Pull request by edouardpoitras**
Found myself needing the ability to extract all objects from a PBF file.
I also needed the list of coordinates for each Way and Relation.
Created this PR in case someone else needed this functionality.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -38,7 +38,9 @@ docker run \
 
 ### Retrieve Objects
 
-You have to specify a query via `--tags` or `-t`, the syntax is rather simple:
+By default the `objects` command will extract all objects from the pbf file provided. The centroid and bounds of Way and Relation objects are provided by default, however the `--retain-coordinates` or `-r` flag will also expose the list of coordinates for each object (similar to the `streets` command).
+
+You can specify a query via `--tags` or `-t`, the syntax is rather simple:
```
Comment (mkulke) [inline Readme.md]:
```suggestion
Centroid and bounds of Way and Relation objects are provided by default, however the `--retain-coordinates` or `-r` flag will also expose the list of coordinates for each object (similar to the `streets` command).

By default the `objects` command will extract all objects. To select a set of objects by their tags you can specify a query via `--tags` or `-t`, the syntax is rather simple:
```

File: src/lib/items.rs [hunk 2]
```diff
@@ -60,17 +60,25 @@ pub mod osm {
...
 
     impl Way {
-        pub fn new(id: i64, tags: Tags, coordinates: &[(f64, f64)]) -> Self {
+        pub fn new(id: i64, tags: Tags, coordinates: &[(f64, f64)], retain_coordinates: bool) -> Self {
             let (centroid, bounds) = get_geo_info(coordinates);
+            let coordinates = if retain_coordinates {
+                Some(coordinates.to_vec())
+            } else {
+                None
+            };
```
Comment (mkulke) [inline src/lib/items.rs]:
```suggestion
            let coordinates = retain_coordinates.then(|| coordinates.into());
```

File: src/lib/items.rs [hunk 3]
```diff
@@ -83,17 +91,25 @@ pub mod osm {
...
 
     impl Relation {
-        pub fn new(id: i64, tags: Tags, coordinates: &[(f64, f64)]) -> Self {
+        pub fn new(id: i64, tags: Tags, coordinates: &[(f64, f64)], retain_coordinates: bool) -> Self {
             let (centroid, bounds) = get_geo_info(coordinates);
+            let coordinates = if retain_coordinates {
+                Some(coordinates.to_vec())
+            } else {
+                None
+            };
```
Comment (mkulke) [inline src/lib/items.rs]:
```suggestion
            let coordinates = retain_coordinates.then(|| coordinates.into());
```

File: src/main.rs [hunk 4]
```diff
@@ -45,9 +47,13 @@ fn main() -> Result<(), Box<dyn Error>> {
     let file = File::open(args.path)?;
 
     match args.cmd {
-        Command::Objects { tags } => {
-            let groups = filter::parse(&tags);
-            let objects = objects(file, &groups)?;
+        Command::Objects { tags, retain_coordinates } => {
+            let objects = if tags.is_some() {
+                let groups = filter::parse(&tags.unwrap());
```
Comment (mkulke) [inline src/main.rs]:
```suggestion
            let objects = if let Some(tags) = tags {
                let groups = filter::parse(&tags);
```

**Review by mkulke — CHANGES_REQUESTED** [review]
thanks for the PR, some minor nitpicks. Also the serialization code looks like a lot of ceremony now, it probably needs to be refactored. but that's out of the scope of that PR.

**Comment by edouardpoitras** [comment]
Very reasonable - committed all proposed changes.

**Review by mkulke — APPROVED** [review]

**Comment by edouardpoitras** [comment]
Thank you kindly.

**Pull request merged**